coding-the-legislation/reforms.md

```diff
@@ -35,28 +38,79 @@ A reform that modifies a formula (such as our `income_tax_reform` example) is ca
 
 ### Parametric reforms
 
-Unlike *structural reforms*, *parametric reforms* do not directly modify any formula: they apply changes to legislation parameters only.
+A reform that apply changes to legislation parameters is called a *parametric reforms*. 
+
+> Note that a reform can be both structural and parametric, modifying and/or adding variables *and* parameters.
 
-To modify the legislation parameters in the reform, you can call the method `self.modify_legislation_json`, which takes a function as a parameter.
+To modify the legislation parameters in the reform, you can call the method `self.modify_legislation`, which takes a function as a parameter.
 
-This function defines the modifications you want to apply to the legislation. It takes as a parameter a copy of the reference tax and benefit system `legislation_json`. You can then modify and return this `legislation_json`.
+This function defines the modifications you want to apply to the legislation. It takes as a parameter a copy of the reference tax and benefit system parameters: `legislation`. You can then modify and return this `legislation`.
 
-For instance:
+#### Update the value of a parameter
 
 ```python
-def modify_legislation_json(reference_legislation_json_copy):
-    reference_legislation_json_copy['children']['minimum_wage']['values'][0]['value'] = 15
-    return reference_legislation_json_copy
+def modify_legislation(legislation):
+    legislation.tax_on_salary.scale[1].threshold.update(period=reform_period, value=new_value)
+    return legislation
 
 
 class increase_minimum_wage(Reform):
     name = u'Increase the minimum wage'
 
     def apply(self):
-        self.modify_legislation_json(modifier_function = modify_legislation_json)
+        self.modify_legislation(modifier_function = modify_legislation)
+```
+
+#### Add new parameters
+
+You can add new parameters from a directory containing YAML files, similarly to the [parameters of the original tax benefit system](legislation_parameters.md).
+
+```python
```

Comment:
I tried to remove some useless speech that made it unclear. If it is still not clear we should discuss it together.